packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -2122,71 +2098,47 @@ class _ScrollUnderFlexibleSpace extends StatelessWidget {
         config = _LargeScrollUnderFlexibleConfig(context);
     }
 
-    late final Widget? collapsedTitle;
     late final Widget? expandedTitle;
     if (title != null) {
-      collapsedTitle = config.collapsedTextStyle != null
-        ? DefaultTextStyle(
-            style: config.collapsedTextStyle!.copyWith(color: foregroundColor ?? appBarTheme.foregroundColor),
-            child: title!,
-          )
-        : title;
+      final TextStyle? expandedTextStyle = titleTextStyle
+        ?? appBarTheme.titleTextStyle
+        ?? config.expandedTextStyle?.copyWith(
+             color: foregroundColor ?? appBarTheme.foregroundColor ?? defaults.foregroundColor,
+           );
       expandedTitle = config.expandedTextStyle != null
         ? DefaultTextStyle(
-            style: config.expandedTextStyle!.copyWith(color: foregroundColor ?? appBarTheme.foregroundColor),
+            style: expandedTextStyle!,
             child: title!,
           )
         : title;
     }
 
-    late final bool centerTitle;
-    {
-      bool platformCenter() {
-        switch (theme.platform) {
-          case TargetPlatform.android:
-          case TargetPlatform.fuchsia:
-          case TargetPlatform.linux:
-          case TargetPlatform.windows:
-            return false;
-          case TargetPlatform.iOS:
-          case TargetPlatform.macOS:
-            return true;
-        }
+    EdgeInsetsGeometry expandedTitlePadding() {
+      final EdgeInsets padding = config.expandedTitlePadding!.resolve(Directionality.of(context));
+      if (bottomHeight > 0) {
+        return EdgeInsets.fromLTRB(padding.left, 0, padding.right, bottomHeight);
       }
-      centerTitle = centerCollapsedTitle ?? appBarTheme.centerTitle ?? platformCenter();
-    }
-
-    EdgeInsetsGeometry effectiveCollapsedTitlePadding = EdgeInsets.zero;
-    if (hasLeading && leadingWidth == null) {
-      effectiveCollapsedTitlePadding = centerTitle
-        ? config.collapsedCenteredTitlePadding!
-        : config.collapsedTitlePadding!;
-    } else if (hasLeading && leadingWidth != null) {
-      effectiveCollapsedTitlePadding = EdgeInsetsDirectional.only(start: leadingWidth!);
+      // print(config.expandedTextStyle.toString());
```

Comment:
```suggestion
```

Left over debugging? :)

Reply:
Ooops

Reply:
Resolved conflict from https://github.com/flutter/flutter/pull/123256 and remove print (thanks for catching that)